egs/librispeech/ASR/conformer_ctc/transformer.py

```diff
@@ -179,10 +180,15 @@ def forward(
               memory_key_padding_mask for the decoder. Its shape is (N, T).
               It is None if `supervision` is None.
         """
-        if self.use_feat_batchnorm:
+        if (
+            isinstance(self.use_feat_batchnorm, bool)
+            and self.use_feat_batchnorm
+        ):
             x = x.permute(0, 2, 1)  # (N, T, C) -> (N, C, T)
             x = self.feat_batchnorm(x)
             x = x.permute(0, 2, 1)  # (N, C, T) -> (N, T, C)
+        if isinstance(self.use_feat_batchnorm, float):
```

Comment:
```python3
if self.use_feat_batchnorm:
            x = x.permute(0, 2, 1)  # (N, T, C) -> (N, C, T)
            x = self.feat_batchnorm(x)
            x = x.permute(0, 2, 1)  # (N, C, T) -> (N, T, C)
else:
            x *= self.use_feat_batchnorm
```

Also, we can add a check in the constructor
```python3
assert  isinstance(use_feat_batchnorm, (bool, float))
```

Reply:
I think `if self.use_feat_batchnorm` returns true if `self.use_feat_batchnorm` is a non-zero float. I will add the assertion and keep the if statements the same as before, it is more clear.

Reply:
I meant
```python
if self.use_feat_batchnorm is True
```

Reply:
We scale the features only when `use_feat_batchnorm` is a float. we have to test  whether `use_feat_batchnorm` is a float or not (i.e. it can be False).